Migration script/openIMIS migration latest.sql

```diff
@@ -8106,9 +7833,269 @@ SELECT F.FamilyId,F.InsureeId, I.CHFID , F.LocationId, F.Poverty
+			ELSE
+				IF @isClaimAdminOptional = 1
+					BEGIN
+						SELECT @ClaimAdminId = ClaimAdminId FROM tblClaimAdmin WHERE ClaimAdminCode = @ClaimAdmin AND ValidityTo IS NULL
+					END
+
+		BEGIN TRAN CLAIM
+			INSERT INTO tblClaim(InsureeID,ClaimCode,DateFrom,DateTo,ICDID,ClaimStatus,Claimed,DateClaimed,Explanation,AuditUserID,HFID,ClaimAdminId,ICDID1,ICDID2,ICDID3,ICDID4,VisitType,GuaranteeId)
+						VALUES(@InsureeID,@ClaimCode,@StartDate,@EndDate,@ICDID,2,@Total,@ClaimDate,@Comment,-1,@HFID,@ClaimAdminId,@ICDID1,@ICDID2,@ICDID3,@ICDID4,@VisitType,@GuaranteeId);
+
+			SELECT @ClaimID = SCOPE_IDENTITY();
+
+			;WITH PLID AS
+			(
+				SELECT PLID.ItemId, PLID.PriceOverule
+				FROM tblHF HF
+				INNER JOIN tblPLItems PLI ON PLI.PLItemId = HF.PLItemID
+				INNER JOIN tblPLItemsDetail PLID ON PLID.PLItemId = PLI.PLItemId
+				WHERE HF.ValidityTo IS NULL
+				AND PLI.ValidityTo IS NULL
+				AND PLID.ValidityTo IS NULL
+				AND HF.HFID = @HFID
+			)
+			INSERT INTO tblClaimItems(ClaimID,ItemID,QtyProvided,PriceAsked,AuditUserID)
+			SELECT @ClaimID, I.ItemId, T.ItemQuantity, COALESCE(NULLIF(T.ItemPrice,0),PLID.PriceOverule,I.ItemPrice)ItemPrice, -1
+			FROM #tblItem T
+			INNER JOIN tblItems I  ON T.ItemCode COLLATE DATABASE_DEFAULT = I.ItemCode COLLATE DATABASE_DEFAULT AND I.ValidityTo IS NULL
+			LEFT OUTER JOIN PLID ON PLID.ItemID = I.ItemID
+
+			SELECT @TotalItems = SUM(PriceAsked * QtyProvided) FROM tblClaimItems
+						WHERE ClaimID = @ClaimID
+						GROUP BY ClaimID
+
+			;WITH PLSD AS
+			(
+				SELECT PLSD.ServiceId, PLSD.PriceOverule
+				FROM tblHF HF
+				INNER JOIN tblPLServices PLS ON PLS.PLServiceId = HF.PLServiceID
+				INNER JOIN tblPLServicesDetail PLSD ON PLSD.PLServiceId = PLS.PLServiceId
+				WHERE HF.ValidityTo IS NULL
+				AND PLS.ValidityTo IS NULL
+				AND PLSD.ValidityTo IS NULL
+				AND HF.HFID = @HFID
+			)
+			INSERT INTO tblClaimServices(ClaimId, ServiceID, QtyProvided, PriceAsked, AuditUserID)
+			SELECT @ClaimID, S.ServiceID, T.ServiceQuantity,COALESCE(NULLIF(T.ServicePrice,0),PLSD.PriceOverule,S.ServPrice)ServicePrice , -1
+			FROM #tblService T
+			INNER JOIN tblServices S ON T.ServiceCode COLLATE DATABASE_DEFAULT = S.ServCode COLLATE DATABASE_DEFAULT AND S.ValidityTo IS NULL
+			LEFT OUTER JOIN PLSD ON PLSD.ServiceId = S.ServiceId
+
+						SELECT @TotalServices = SUM(PriceAsked * QtyProvided) FROM tblClaimServices
+						WHERE ClaimID = @ClaimID
+						GROUP BY ClaimID
+
+						UPDATE tblClaim SET Claimed = ISNULL(@TotalItems,0) + ISNULL(@TotalServices,0)
+						WHERE ClaimID = @ClaimID
+
+		COMMIT TRAN CLAIM
+
+
+		SELECT @ClaimID  = IDENT_CURRENT('tblClaim')
+
+		IF @ByPassSubmit = 0
+		BEGIN
+			DECLARE @ClaimRejectionReason INT
+			EXEC @ClaimRejectionReason = uspSubmitSingleClaim -1, @ClaimID,0
+			IF @ClaimRejectionReason <> 0
+				SELECT @ClaimRejected = 1
+		END
+
+	END TRY
+	BEGIN CATCH
+		IF @@TRANCOUNT > 0
+			ROLLBACK TRAN CLAIM
+			SELECT ERROR_MESSAGE()
+		RETURN -1
+	END CATCH
+
+	RETURN 0
+END
```

Comment:
Is te GO missing after this SP?

Reply:
I Fixed the issue. It is not required for the script to run, but the next append to the migration will need this.